src/hooks/street-view-panorama.ts

```diff
@@ -0,0 +1,72 @@
+import {useMemo, useState} from 'react';
+
+import {useGoogleMap} from './map-instance';
+
+interface StreetViewProps {
+  divElement?: HTMLElement | null;
+  position?: google.maps.LatLng | google.maps.LatLngLiteral;
+  heading?: number;
+  pitch?: number;
+  zoom?: number;
+}
+/**
+ * Hook to get Street View Panorama
+ */
+export const useStreetView = (
+  props: StreetViewProps
+): google.maps.StreetViewPanorama | null => {
+  const {divElement, position, heading, pitch, zoom} = props;
+
+  // Get map instance
+  const {map} = useGoogleMap();
+
+  const [panorama, setPanorama] =
+    useState<google.maps.StreetViewPanorama | null>(null);
+
+  // Creates a Street View instance
+  const streetViewPanorama =
+    useMemo<google.maps.StreetViewPanorama | null>(() => {
+      // Wait for Google Maps API to be initialized
+      if (!map) {
+        return null;
+      }
+
+      if (!divElement) {
+        const newPanorama = map.getStreetView();
+        newPanorama.setPov({heading, pitch});
+
+        if (position) {
+          newPanorama.setPosition(position);
+        }
+
+        if (zoom) {
+          newPanorama.setZoom(zoom);
+        }
+
+        newPanorama.setVisible(true);
```

Comment:
Shouldn't we leave this to the developer, whether he/she wants to show the street view panorama right away or not? I think it's not a very common use case to start with a street view panorama, but a more common use case to start with a map and open a street view image for example if the user clicks somewhere on the map.
If this makes sense for you too, please remove this line of code and add a hint to the README, that the street view panorama is only visible after calling `panorama.setVisible(true)`.